recipe/meta.yaml

```diff
@@ -1,28 +1,58 @@
 {% set name = "CoolProp" %}
 
+# WARNING external resources may need update as well
 {% set version = "6.2.1" %}
-{% set sha256 = "d5b85af023ffc93d3f85770dee571712f3858f72579e89a7d13b2f796d933732" %}
+{% set sha256 = "e94a65e5bbf150e4010816160f66b2095dc70b598461bb1fa664bb5791922c93" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  - url: https://github.com/CoolProp/{{ name }}/archive/v{{ version }}.tar.gz
+    sha256: {{ sha256 }}
+    patches:
+      - setup.patch
+      - CMakeLists.patch
+  # External dependencies
+  # See https://github.com/CoolProp/CoolProp/tree/master/externals
+  - url: https://github.com/catchorg/Catch2/archive/v1.12.2.tar.gz
+    sha256: 27bf01d20692e534a8963f96c6ca797df2b4ba6551db0379510c376558d75e3c
```

Comment:
These improvements look very good, @fcollonval 

I'm just worried about the maintainability of this part: there are too many submodules here, it seems burdensome to update all versions and hashes, and I don't know if the regro-bot automatically updates these. Do you know if there's something that can be done to improve this? Perhaps falling back to using `git:` source? I've seem that there is an open discussion about this here: https://github.com/conda-forge/conda-forge.github.io/issues/619

Reply:
I cannot agree more. But unfortunately I didn't find another way. And I'm pretty sure the bot is not able to handle that.

Two remarks:

* 6.3.0 has only one change in the external dependencies. So it looks they are not the kind of jumping to the latest version available
* Some of those dependencies are available as conda packages. So the next step could be to build coolprop with two *outputs*; one being the cpp code stored in dynamic linked library and the second being the python library linked to it.

Reply:
> Some of those dependencies are available as conda packages. So the next step could be to build coolprop with two outputs; one being the cpp code stored in dynamic linked library and the second being the python library linked to it.

I agree, this would better be done collaborating with their [upstream repo](https://github.com/coolprop/coolprop), they have Travis CI configured, perhaps there could be another build that uses conda and takes dependencies from conda-forge.

Reply:
At second thought, it may not be such a great idea if the third party dependencies needs to be pinned to an "old" version - in this case I think of `Catch2` for example (kept at 1.x).